### ISSUE TITLE
New Rule for dangerous type casts

### ISSUE THREAD

**Issue by tdellmann** [mgechev/revive#888]
A dynamic type cast like `x := something(*somethingElse[])` can cause a panic.  Because this can be extremely dangerous there should be a lint rule for it.

This rule should trigger if 
- a type ist dynamicly casted to a single value
- a type is dynamicly casted and the second response value (ok) is ignored
  - the first result can be checked for nil - so this might be nice to configure
- a dynamic type cast happens in a return statement

The alternative would be to always `panic.recover()` in bigger projects and just live with something as unsafe like
```golang
return ctx.Value("myCoolErrorChan").(chan err)
```
